### ISSUE TITLE
Possible to detect a "prefetch" visit in the backend?

### ISSUE THREAD

**Issue by andregoldstein** [hotwired/turbo#1327]
I'm in the process of setting up "user tracking" for an internal tool, part of this sends the most recent path to Pusher. 

However, I'm not aware of any way of detecting if an action has been initiated as part of a "prefetch" visit or not.

It would appear that `HTTP_X_SEC_PURPOSE` is set to `prefetch` even on a full click event (I suppose it is an event that was prefetched before committing to the click).

Unless I can differentiate between the two I guess the options remaining are to disable prefetching with `<meta name="turbo-prefetch" content="false">` or to implement this in the frontend when the page changes

**Comment by Ambient-Impact** [comment]
I've been pondering on this particular problem, and the only solution I can think of is to make a second request to the back-end if/when a prefetch is used as navigation, i.e. user actually clicks the link and a visit occurs, to actually log it as a "real" event.

One crude but relatively easy solution may be to have an event handler that simply sends a second copy of the fetch request with all the same headers except for the one indicating a prefetch to essentially ping the back-end. You'd want to be careful with that, and I can see it failing if you're using nonces, etc., but it may work.

**Comment by andregoldstein** [comment]
Thanks, some good workarounds. 

It's a shame that a header can't be sent on prefetch only visits - it must be useful information for all sorts of reasons - analytics etc?

**Comment by Ambient-Impact** [comment]
I agree it would be very useful, but because of how the prefetch works - by the time you click or decide not to, it's already been sent to the server - the server can't know at the time it receives the request whether the response it sends is viewed at all because you haven't clicked yet.

Under vanilla HTTP conditions, you probably have no choice but to make a second background request to let the server know that a link was clicked for real.

It might be possible to do it all in one long lived connection with WebSockets or Mercure, but I haven't had a chance to experiment with those.

**Comment by andregoldstein** [comment]
Thanks, I had assumed it would be possible for Turbo to send a custom header to signal a prefetch when this request is made? I'd thought Turbo would be in control of this request?

**Comment by Ambient-Impact** [comment]
Like I said, it's because the click happens _after_ the request is sent to the server, not the other way around. Like you said in the issue summary, Turbo does provide a header, but it's already been sent with that header when you click.
